src/webglimpse/timeline/timeline_events_row.ts

```diff
@@ -705,39 +710,87 @@ module Webglimpse {
                 '                                                                ',
                 '  attribute vec2 a_XyFrac;                                      ',
                 '  attribute vec4 a_Color;                                       ',
-                '                                                                ',
+                '  attribute vec2 a_relativeXy;                                   ',
+                '  attribute float a_fillPattern;                                    ',     
+                '                                                                ',                
                 '  varying vec4 v_Color;                                         ',
+                '  varying vec2 v_relativeXy;                                     ', 
+                '  varying float v_fillPattern;                                     ',                                               
                 '                                                                ',
                 '  void main( ) {                                                ',
                 '      gl_Position = vec4( ( -1.0 + 2.0*a_XyFrac ), 0.0, 1.0 );  ',
                 '      v_Color = a_Color;                                        ',
+                '      v_relativeXy = a_relativeXy;                                ',
+                '      v_fillPattern = a_fillPattern;                                ',                
                 '  }                                                             ',
                 '                                                                '
             );
 
-        var program = new Program( xyFrac_vColor_VERTSHADER, varyingColor_FRAGSHADER );
+
+         var fillPattern_FRAGSHADER = Webglimpse.concatLines(
+                '                                                                           ',
+				'  precision highp float;                                                   ',
+				'  varying vec4 v_Color;                                                    ',
+   				'  varying vec2 v_relativeXy;                                                ',
+   				'  varying float v_fillPattern;                                              ',                   
```

Comment:
No you're right. Since they can be different for each event it's correct as is. I just wasn't thinking it through completely.

However, since they value the will be the same for all the vertices, I believe we can make it more efficient by using 'varying flat' to tell OpenGL it doesn't need to interpolate the value assigned to the fragment (it's possible OpenGL ES doesn't allow the flat keyword, haven't checked what version of GLSL it requires).